### ISSUE TITLE
No module named 'decorator'

### ISSUE THREAD

**Issue by robshep** [circstat/pycircstat#64]
I think the import should be **decorators**

```
File "arc.py", line 5, in <module>
    import pycircstat
  File "/[...]/.venv/lib/python3.7/site-packages/pycircstat/__init__.py", line 6, in <module>
    from .descriptive import *
  File "/[...]/.venv/lib/python3.7/site-packages/pycircstat/descriptive.py", line 8, in <module>
    from decorator import decorator
ModuleNotFoundError: No module named 'decorator'
```

**Comment by fabiansinz** [comment]
`decorator` is correct. It has nothing to do with the local file `decorators` if you look in how it is used in the file. I am using python 3.7.3 and just imported `decorator.decorator` without any problem. Can you try that import from your local python prompt?

**Comment by robshep** [comment]
I was importing descriptors.

No longer using this lib in favour of R’s circular stats library.
Thanks

Rob

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by robshep** [comment]
Just checked this out and in python 3.7.2

>>> import decorator.decorators
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'decorator'

I'm working in a python3 venv and I've uninstalled and re-installed this module but it doesn't bring any dependencies.   I don't have any experience with python modules, but maybe a problem for others. 

Thanks

Rob

**Comment by fabiansinz** [comment]
My current guess is that there is something wrong with your venv. If you installed the module you should be able to import it. Does `import decorator` work?

**Comment by robshep** [comment]
No, I can't `import decorator` as above. 

It's a new venv just made to try this lib.  

I've recreated it here, if you want to follow:

```
rob@graviton-161 circular$ cd /tmp
rob@graviton-161 tmp$ mkdir test_pycircstat
rob@graviton-161 tmp$ cd test_pycircstat/
rob@graviton-161 test_pycircstat$ /usr/local/bin/python3.7 -V
Python 3.7.2
rob@graviton-161 test_pycircstat$ /usr/local/bin/python3.7 -m venv .venv
rob@graviton-161 test_pycircstat$ source .venv/bin/activate
(.venv) rob@graviton-161 test_pycircstat$ which pip
/private/tmp/test_pycircstat/.venv/bin/pip
(.venv) rob@graviton-161 test_pycircstat$ pip install pycircstat
Collecting pycircstat
  Using cached https://files.pythonhosted.org/packages/60/28/a154b18dfa5b1d767a2a9b99977fa94285bd8defa6d5eb553095bda3aa70/pycircstat-0.0.2-py3-none-any.whl
Installing collected packages: pycircstat
Successfully installed pycircstat-0.0.2
You are using pip version 18.1, however version 19.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(.venv) rob@graviton-161 test_pycircstat$ which python
/private/tmp/test_pycircstat/.venv/bin/python
(.venv) rob@graviton-161 test_pycircstat$ python -V
Python 3.7.2
(.venv) rob@graviton-161 test_pycircstat$ python
Python 3.7.2 (default, Feb 12 2019, 08:16:38)
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import decorator
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'decorator'
>>> import pycircstat
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/tmp/test_pycircstat/.venv/lib/python3.7/site-packages/pycircstat/__init__.py", line 6, in <module>
    from .descriptive import *
  File "/private/tmp/test_pycircstat/.venv/lib/python3.7/site-packages/pycircstat/descriptive.py", line 8, in <module>
    from decorator import decorator
ModuleNotFoundError: No module named 'decorator'
```

Apologies if it's something weird on my specific system.  
It would be nice to return to this lib later ;) 

Rob

**Comment by fabiansinz** [comment]
Do you have docker installed? Maybe it would be work trying the import in a `python3.7` image.

**Comment by robshep** [comment]
it works if I install decorator manually.

should the pip not fetch it?

**Comment by robshep** [comment]
if not then please consider updating the readme installation steps to install dependencies manually.

thanks

**Comment by fabiansinz** [comment]
Ok that's interesting. I thought you did try to install it manually before. Quick question, how did you install `pycircstat` in the first place? Technically, `decorator` is in the dependencies mentioned in the `setup.py` so it should be installed.

**Comment by robshep** [comment]
`pip`

See: https://github.com/circstat/pycircstat/issues/64#issuecomment-496190301

**Comment by fabiansinz** [comment]
Ok, I just verified that this is missing in the pip version and needs to be installed. Thanks for the feedback.

**Comment by avdaranyi** [comment]
This would fix it: #69
